drizzlepac/devutils/search_skyframes.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python
+
+""" This script allows the user to query a master observations .csv file by any combination of exposure name,
+    skycell name, instrument/detector, and/or filter name. The results are printed the screen and optionally
+    written to a user-specified output .csv file. The results will be sorted first by skycell name, then by
+    instrument/detector, filter name and finally by image name."""
+
+import argparse
+from datetime import datetime
+
+import sys
+
+from astropy.io import fits
+import pandas as pd
+
+from drizzlepac.haputils import cell_utils
+from drizzlepac.haputils import make_poller_files
+
+
+# ------------------------------------------------------------------------------------------------------------
+
+def augment_results(results):
+    """Add additional searchable columns to query results. Columns added are date-obs and  full flt/flc fits
+    file path
+
+    Parameters
+    ----------
+    results : pandas.DataFrame
+        query results to augment
+
+    Returns
+    -------
+    results : pandas.DataFrame
+        Augmented query results
+    """
+    dateobs_list = []
+    path_list = []
+
+    # populate dateobs_list, path_list
+    for idx in results.index:
+        rootname = results.exposure[idx]
+        imgname = make_poller_files.locate_fitspath_from_rootname(rootname)
+        dateobs_list.append(fits.getval(imgname, "DATE-OBS"))
+        path_list.append(imgname)
+
+    # Add 'dateobs' (as datetime objects) and 'filename' columns to results dataframe.
+    results['dateobs'] = dateobs_list
+    results['dateobs'] = pd.to_datetime(results.dateobs)
+    results['filename'] = path_list
+    return results
+
+
+# ------------------------------------------------------------------------------------------------------------
+
+def make_search_string(arg_dict):
+    """Create search string that will be used to query observation tables
+
+    Parameters
+    ----------
+    arg_dict: dict
+        dictionary with the search arguments. They are as follows: 'exposure': Image name,
+        'skycell': Skycell name, 'config': Instrument/Detector name (Uppercase, separated by a "/"
+        (e.g. WFC3/IR)), 'spec': Filter name. lowercase.
```

Comment:
It may be helpful for future users to know how to provide multi-filter use-cases, since ACS can use 2 filters at once.

Reply:
issue resolved by commit [04379d8].(https://github.com/spacetelescope/drizzlepac/pull/964/commits/04379d8ac5eefe2d50335d3da7cc9c73824050b1)